[conex] Disable the DUMP() macro and all other printing

Dumping to cout and cerr from a library is completely unsuitable; only
applications should interact with the console.

Reasoning for not upstreaming this patch: Drake-specific build option.

--- conex/debug_macros.h
+++ conex/debug_macros.h
@@ -2,8 +2,6 @@
 #include <array>
 #include <cassert>
 #include <chrono>
-#include <iomanip>
-#include <iostream>
 #include <vector>
 
 namespace conex {
@@ -8,8 +8,6 @@
 #define DUMP(x)                                              \
-  std::cerr << __FILE__ << " line " << __LINE__ << std::endl \
-            << #x ":" << std::endl                           \
-            << x << std::endl;
+  do { (void)(x); } while (0);  /* Removed for Drake. */
 
 #ifndef CONEX_VERBOSE
-#define CONEX_VERBOSE 1
+#define CONEX_VERBOSE 0
 #endif
@@ -61,6 +61,7 @@
 
 #endif
 
+#if 0  // Removed for Drake.
 template <typename T1, typename T2>
 inline std::ostream& operator<<(std::ostream& os, const std::pair<T1, T2>& P) {
   os << P.first << " " << P.second << "\n\n";
@@ -84,6 +85,7 @@
   }
   return os;
 }
+#endif
 
 }  // namespace conex

--- conex/elimination_ordering.cc
+++ conex/elimination_ordering.cc
@@ -9,7 +9,6 @@
 //
 #include "conex/elimination_ordering.h"
 
-#include <iostream>
 #include <vector>
 
 namespace conex {

--- conex/error_checking_macros.h
+++ conex/error_checking_macros.h
@@ -1,4 +1,3 @@
-#include <iostream>
 #include "conex/error_codes.h"
 
 namespace conex {
@@ -14,7 +14,7 @@
 #else
 #define CONEX_DEMAND(x, msg)                                                   \
   if (!(x)) {                                                                  \
-    std::cerr << __FILE__ << " line " << __LINE__ << ": " << msg << std::endl; \
+    (void)(msg);  /* Removed for Drake. */                                     \
     return 1;                                                                  \
   }
 #endif

--- conex/supernodal_solver.cc
+++ conex/supernodal_solver.cc
@@ -1,7 +1,6 @@
 #include "conex/supernodal_solver.h"
 #include "conex/clique_ordering.h"
 
-#include <iostream>
 #include <map>
 
 #include <Eigen/Dense>

--- conex/workspace.h
+++ conex/workspace.h
@@ -1,5 +1,4 @@
 #pragma once
-#include <iostream>
 #include <memory>
 #include <vector>
 
@@ -85,12 +85,14 @@
     return o.max_iter_ * num_items + 2;
   }
 
+#if 0  // Removed for Drake.
   friend void print(const WorkspaceStats& o) {
     for (int i = 0; i < o.max_iter_; i++) {
       std::cout << "sqrt_inv_mu, norm_inf_d" << o.sqrt_inv_mu[i] << ", "
                 << o.norm_inf_d[i] << "\n";
     }
   }
+#endif
 
   bool IsInitialized() { return initialized; }
 
